packages/core/test/epochTree.test.ts

```diff
@@ -110,37 +111,32 @@ describe('Epoch tree', function () {
         const leaves = [] as any
 
         for (const epk of epochKeys) {
-            let posRep = 0
-            let negRep = 0
-            let graffiti = BigInt(0)
-            let timestamp = 0
+            const data = Array(userState.sync.settings.fieldCount).fill(
+                BigInt(0)
+            )
             for (let i = 0; i < 5; i++) {
-                const newPosRep = Math.floor(Math.random() * 10)
-                const newNegRep = Math.floor(Math.random() * 10)
-                const newGraffiti = genRandomSalt()
+                const fieldIndex = Math.floor(
+                    Math.random() * (userState.sync.settings.sumFieldCount + 1)
+                )
+                const val = Math.floor(Math.random() * 10000000000000)
                 // now submit the attestation from the attester
                 const { timestamp: newTimestamp } = await unirepContract
                     .connect(attester)
-                    .submitAttestation(
-                        epoch,
-                        epk,
-                        newPosRep,
-                        newNegRep,
-                        newGraffiti,
-                        {
-                            gasLimit: 1000000,
-                        }
-                    )
+                    .attest(epk, epoch, fieldIndex, val, {
+                        gasLimit: 1000000,
+                    })
                     .then((t) => t.wait())
                     .then(({ blockNumber }) =>
                         ethers.provider.getBlock(blockNumber)
                     )
-                posRep += newPosRep
-                negRep += newNegRep
-                graffiti = newGraffiti
-                timestamp = newTimestamp
+                if (fieldIndex < userState.sync.settings.sumFieldCount) {
+                    data[fieldIndex] += BigInt(val)
```

Comment:
i think it's unnecessary because the sum fields use modular addition in the field like [here](https://github.com/Unirep/Unirep/pull/322/files#diff-bd4415896551787d4d50aa8e8bf1cf39f56c05251c9f8fdb9b1746aca509ec02R249)

Reply:
yeah it is handled on-chain
but I think generating a test that does not handle edge cases is not good
it can be like UserState
```ts
    data[fieldIndex] = data[fieldIndex] + (val % F)
```